lib/gameboard/board.rb

```diff
@@ -0,0 +1,495 @@
+require_relative './cell'
+
+module Gameboard
+  # The Main Board Model
+
+  class Board
+    # Public: Returns the height of the gameboard.
+    attr_reader :height
+
+    # Public: Returns the width of the gameboard.
+    attr_reader :width
+
+    include Enumerable
+
+    # Public: Initialize a gameboard.
+    #
+    # height - The number of horizontal rows on the gameboard. Not passing a
+    #          height will raise an ArgumentError unless a preset board is
+    #          passed
+    # width  - The number of columns on the gameboard. Not passing a
+    #          width will raise an ArgumentError unless a preset board is
+    #          passed
+    # cells  - An optional value, preferably a String in most cases, that
+    #          changes the default cell value.
+    # preset - An optional 2D Array to load a preset gameboard. Saved games
+    #          should be created in rows from the top down.
+    #
+    # Examples
+    #
+    #   board = Board.new(cells: "O", width: 10, height: 8)
+    #   # =>     -----------------------------------------
+    #          0 | O | O | O | O | O | O | O | O | O | O |
+    #            -----------------------------------------
+    #          1 | O | O | O | O | O | O | O | O | O | O |
+    #            -----------------------------------------
+    #          2 | O | O | O | O | O | O | O | O | O | O |
+    #            -----------------------------------------
+    #          3 | O | O | O | O | O | O | O | O | O | O |
+    #            -----------------------------------------
+    #          4 | O | O | O | O | O | O | O | O | O | O |
+    #            -----------------------------------------
+    #          5 | O | O | O | O | O | O | O | O | O | O |
+    #            -----------------------------------------
+    #          6 | O | O | O | O | O | O | O | O | O | O |
+    #            -----------------------------------------
+    #          7 | O | O | O | O | O | O | O | O | O | O |
+    #            -----------------------------------------
+    #            - 0 - 1 - 2 - 3 - 4 - 5 - 6 - 7 - 8 - 9 -
+    #
+    #
+    #   example_arr = [
+    #                  ["1", "2", "3"], #=> coordinates [[0,2], [1,2], [2,2]]
+    #                  ["A", "B", "C"], #=> coordinates [[0,1], [1,1], [2,1]]
+    #                  ["X", "Y", "Z"]  #=> coordinates [[0,0], [1,0], [2,0]]
+    #                 ]
+    #
+    #   board = Board.new(preset: example_array)
+    #           #=>     -------------
+    #                 0 | 1 | 2 | 3 |
+    #                   -------------
+    #                 1 | A | B | C |
+    #                   -------------
+    #                 2 | X | Y | Z |
+    #                   -------------
+    #                   - 0 - 1 - 2 -
+    #
+    #
+    def initialize(height:false, width: false, cells: false, preset: false)
```

Comment:
so I did `height: false, width: false` instead of `height:, width:` so that i could not raise an error when preset is passed, but still flag them when its not. is there a better way for that situation?